packages/components/src/Chip/Chip.tsx

```diff
@@ -84,10 +86,47 @@ export const Chip = ({
           )}
         </div>
         {suffix}
-      </Tag>
-    </Tooltip>
-  );
-};
+      </>
+    );
+
+    // Use createElement to properly handle the ref typing
+    return (
+      <Tooltip message={tooltipMessage} setTabIndex={false}>
+        {onClick
+          ? React.createElement(
+              "button",
+              {
+                className: classes,
+                onClick: (ev: React.MouseEvent<HTMLButtonElement>) =>
+                  onClick?.(value, ev),
+                tabIndex: disabled ? -1 : tabIndex,
+                onKeyDown,
+                "aria-label": ariaLabel,
+                disabled,
+                role,
+                "data-testid": testID,
+                type: "button",
+                ref,
+              },
+              chipContent,
+            )
+          : React.createElement(
+              "div",
+              {
+                className: classes,
+                tabIndex: disabled ? -1 : tabIndex,
+                onKeyDown,
+                "aria-label": ariaLabel,
+                role,
+                "data-testid": testID,
+                ref,
+              },
+              chipContent,
+            )}
```

Comment:
this isn't a 1:1

we're now missing the `onClick`. even if that's questionable to have for a div, I am not trying to refactor it or remove functionality to that degree right now.

omitting `type: "button"` makes sense but is also not a 1:1

`disabled` is another weird one to apply to a div, but it was doing that before so perhaps I should continue

somewhat surprised the `onClick` isn't upset about the type of the event

Reply:
wait, I'm being silly. of course there's no `onClick` for `div`, the logic of `Tag` specifically only renders a `div` if there is no `onClick` so that part is fine. `type: "button"` also wouldn't make sense for a `div`.

`disabled` is debatable. I suppose it would depend on if we're targeting that attribute for styles, that might be a valid reason to have it on a div.